standardization/src/main/scala/za/co/absa/enceladus/standardization/StandardizationJob.scala

```diff
@@ -303,4 +306,46 @@ object StandardizationJob {
       case Some(rawPathOverride) => rawPathOverride
     }
   }
+
+  /**
+    * Adds metadata about the number of records in raw data by checking Atum's checkpoints first.
+    * If raw record count is not available in checkpoints the method will calculate that count
+    * based on the provided raw dataframe.
+    *
+    * @return The number of records in a checkpoint corresponding to raw data (if available)
+    */
+  private def addRawRecordCountToMetadata(df: DataFrame): Unit = {
+    val checkpointRawRecordCount = getRawRecordCountFromCheckpoints
+
+    val rawRecordCount = checkpointRawRecordCount match {
+      case Some(num) => num
+      case None      => df.count
+    }
+    Atum.setAdditionalInfo(s"raw_record_count" -> rawRecordCount.toString)
+  }
+
+  /**
+    * Gets the number of records in raw data by traversing Atum's checkpoints.
+    *
+    * @return The number of records in a checkpoint corresponding to raw data (if available)
+    */
+  private def getRawRecordCountFromCheckpoints: Option[Long] = {
+    val controlMeasure = Atum.getControMeasure
+
+    val rawCheckpoint = controlMeasure
+      .checkpoints
+      .find(c => c.name.equalsIgnoreCase("raw") || c.workflowName.equalsIgnoreCase("raw"))
```

Comment:
- If Raw is not present I fallback to `df.count` on the initial dataframe. Although not perfect, isn't it the best thing we can do?
- If we write `raw_record_count` metric based on 'Source' checkpoint it will be confusing, won't it?
- Record count can be 0, can't it?
- I'm not sure we should do any validations and sanity checks at this stage. These mismatches can be detected in Statistics/Monitoring. That's what it does already - alerts on mismatches.

Reply:
Ok, since we are going to check controls of all of the checkpoints separately, the source value is probably not required here. Out of 88899 runs, the Source checkpoint is missing in 2841 instances, the Raw is missing in 3600. So the relative difference is not huge. 

I would still recommend to check the value from raw to be non-negative and fallback to count. This way the result will be more reliable and require less checks in monitoring/statistics.

Reply:
Added the sanity check you requested.